[API] New directory for jupyter notebooks

spell00 opened the pull request:
## Brief summary of changes

New Jupyter notebooks to display how the api works.

xlecours commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,254 @@
+   "source": [
+    "### Scripting and parsing json; maybe with python \n",
+    "  \n",
+    "see: [part2 - Python script](https://github.com/xlecours/loris-api-presentation/blob/master/LORIS-API_Part2-Python-script.ipynb)"
```
This should link to the one in aces/Loris

christinerogers replied:
@spell00 could you update this today?

xlecours commented:
@spell00 Remove the outputs of the notebooks before sending to github please.

spell00 commented:
@xlecours I provided the markdowns, but now that I cleared the outputs from the notebooks before sending to Github, the content of the notebooks that I see is very similar to the markdowns. I wonder if it is still necessary? Before clearing the output, I could not see part2 as a notebook in Github, so I understood the necessity of the markdowns, but it seems redundant now. Does it serve another purpose I am not seeing?

xlecours commented:
> @xlecours I provided the markdowns, but now that I cleared the outputs from the notebooks before sending to Github, the content of the notebooks that I see is very similar to the markdowns. I wonder if it is still necessary? Before clearing the output, I could not see part2 as a notebook in Github, so I understood the necessity of the markdowns, but it seems redundant now. Does it serve another purpose I am not seeing?

Maybe @christinerogers can clarify if the markdown can be useful in our readthedoc. Otherwise, it can be removed so we don't have to make sure it is synchronized whit the actual notebook.

spell00 commented:
@xlecours @christinerogers I think everyone agrees the markdowns are redundant, so I removed them. If anyone thinks of a good reason to include them, it can be added again very easily. I added links to the notebooks in Google colab, so that it can be tried directly in the browser. This way, anyone can test the notebooks even if they don't have Jupyter or Loris installed.

xlecours commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,248 @@
+   "source": [
+    "### Scripting and parsing json \n",
+    "  \n",
+    "see: [part2 - Python script](https://github.com/aces/Loris/blob/2020-07-20-jupyterApiPart2/docs/notebooks/LORIS-API_Part2-Python-script.ipynb)"
```
broken link

spell00 replied:
I updated this to a link that will not be broken when the notebooks are merged (see my comment below)

xlecours commented on docs/notebooks/LORIS_API_Part2_Python-script.ipynb:
```diff
@@ -0,0 +1,2159 @@
+    "\n",
+    "    Parameters: baseurl (URL of the api, e.g. https://demo.loris.ca/api/v0.0.3)\n",
+    "\n",
+    "    Returns: projects (Information on all projects)\n",
```
This is not accurate

spell00 replied:
I changed it, but it looks like Github doesn't track Jupyter notebooks very well...

xlecours commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,248 @@
+   "source": [
+    "### Scripting and parsing json \n",
+    "  \n",
+    "see: [part2 - Python script](https://github.com/aces/Loris/blob/2020-07-20-jupyterApiPart2/docs/notebooks/LORIS-API_Part2-Python-script.ipynb)"
```
broken link

spell00 replied:
I updated this to a link that will not be broken when the notebooks are merged (see my comment below)

xlecours commented on docs/notebooks/LORIS_API_Part2_Python-script.ipynb:
```diff
@@ -0,0 +1,2159 @@
+    "    f = open(filename, \"w+b\")\n",
+    "    f.write(bytes(image.content))\n",
+    "\n",
+    "def PUT(url, json_input):\n",
```
Those functions are a nice addition but I think they should return the response instead of printing. There are information in those responses (body, headers) that would be useful to the users.

spell00 replied:
I changed it to return the response. I think you are right, it is more convenient to return the response.

spell00 commented on docs/notebooks/LORIS_API_Part2_Python-script.ipynb:
```diff
@@ -0,0 +1,2159 @@
+    "# Exercise 8 - Dicoms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "ZJhfo1iB_PqE"
+   },
+   "outputs": [],
+   "source": [
+    "# The flag is used to break the loops once an example that contains a \n",
+    "# Dicom file is found\n",
+    "flag = 0\n",
+    "\n",
+    "for candidate in allCandidates['Candidates']:\n",
+    "    candid = candidate['CandID']\n",
+    "    visit_labels = GET('/'.join([baseurl, 'candidates', candid]))['Visits']\n",
+    "    if flag == 1:\n",
+    "        break\n",
+    "    candid = candidate['CandID']\n",
+    "    visit_labels = GET('/'.join([baseurl, 'candidates', candid]))['Visits']\n",
+    "    for visit_label in visit_labels:\n",
+    "        dicoms = GET('/'.join([baseurl, 'candidates', candid, \n",
+    "                               visit_label, 'dicoms']))\n",
+    "        # We only want to get a single valid example of a candidate and visit that\n",
+    "        # has Dicom files that can be used to test the Dicoms endpoints \n",
+    "        if len(dicoms['DicomTars']) > 0:\n",
+    "            flag = 1\n",
+    "            break\n"
+   ]
+  },
```
This block won't be necessary when PR #6775 will be merged. It will add an endpoint listing all Dicoms in a project. It will then be possible to get a complete list of candidates and for which visits there is a Dicom file for a given project. For the moment, it is the only way to get these informations without hardcoding them in the notebook, which is problematic when not executed with raisinbread...

spell00 commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,248 @@
+   "source": [
+    "### Scripting and parsing json \n",
+    "  \n",
+    "see: [part2 - Python script](https://github.com/aces/Loris/blob/main/docs/notebooks/LORIS-API_Part2-Python-script.ipynb)"
```
This link will be valid when the PR is merged to `aces/main`

spell00 commented:
Some commands will fail until PR #6780 is merged. I tested using the branch containing the changes and there is no error.

christinerogers commented:
@driusan requesting final review - this is our last stretch in which @spell00 can readily make changes.  thank you!

spell00 commented:
blocked by #7101

spell00 commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,248 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+   "#LORIS API Tour 1/2\nThis tutorial is also available as a Google colab notebook so you can run it directly from your browser. To access it, click on the button below: <a href=\"https://colab.research.google.com/github/spell00/Loris/blob/2020-07-20-jupyterApiPart2/docs/notebooks/LORIS-API_Part1-HTTP.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\ ...
```
```suggestion
   "#LORIS API Tour 1/2\nThis tutorial is also available as a Google colab notebook so you can run it directly from your browser. To access it, click on the button below: <a href=\"https://colab.research.google.com/github/spell00/Loris/blob/main/docs/notebooks/LORIS-API_Part1-HTTP.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```

spell00 replied:
Sorry for commenting on my own PR, but I want to point out that this link should be changed to the link I am suggesting here. I am not changing it yet so that it can be tested, because the link I am suggesting will only be available once the PR is merged.

spell00 commented on docs/notebooks/LORIS-API_Part1-HTTP.ipynb:
```diff
@@ -0,0 +1,248 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+   "#LORIS API Tour 1/2\nThis tutorial is also available as a Google colab notebook so you can run it directly from your browser. To access it, click on the button below: <a href=\"https://colab.research.google.com/github/aces/Loris/blob/main/docs/notebooks/LORIS-API_Part1-HTTP.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/> ...
```
The link https://colab.research.google.com/github/aces/Loris/blob/main/docs/notebooks/LORIS-API_Part1-HTTP.ipynb won't work until the PR is merged. For now, for testing purposes, one has to use the notebook in this Pr on this branch : https://colab.research.google.com/github/spell00/Loris/blob/2020-07-20-jupyterApiPart2/docs/notebooks/LORIS-API_Part1-HTTP.ipynb

spell00 commented on docs/notebooks/LORIS_API_Part2_Python-script.ipynb:
```diff
@@ -0,0 +1,2149 @@
+    "\n",
+    "This tutorial contains basic examples in Python to demonstrate how to interact with the API.\n",
+    "To run this tutorial, click on `Runtime -> Run all`",
+    "This tutorial is also available as a Google colab notebook so you can run it directly from your browser. To access it, click on the button below: <a href=\"https://colab.research.google.com/github/aces/Loris/blob/main/docs/notebooks/LORIS_API_Part2_Python-script.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
the link https://colab.research.google.com/github/aces/Loris/blob/main/docs/notebooks/LORIS_API_Part2_Python-script.ipynb won't work until the PR is merged. For testing, use https://colab.research.google.com/github/spell00/Loris/blob/2020-07-20-jupyterApiPart2/docs/notebooks/LORIS_API_Part2_Python-script.ipynb

spell00 commented:
@xlecours @driusan It is now passing Travis and ready for final review

driusan reviewed (commented):
I don't know enough about jupyter notebooks to really review this, I'll merge it based on @xlecours approval..
